### ISSUE TITLE
make check of E-Mail address case insensitive

### ISSUE THREAD

**Issue by Hobby-Student** [jstedfast/MimeKit#146]
Sorry for disturbing again. ;)
Don't have time to make a pull request, but my suggestion has just few chars...

I would change the following in "GetCmsRecipient":

from
```cs
...
if (certificate.GetSubjectEmailAddress() == mailbox.Address)
...
```

to
```cs
...
if (certificate.GetSubjectEmailAddress().ToLower() == mailbox.Address.ToLower())
...
```

I don't think that E-Mail addresses are somewhere case sensitive...

**Comment by jstedfast** [comment]
The problem is that the local-part (the part before the @) is server dependent. However, domains can be treated as case insensitive.

**Comment by jstedfast** [comment]
N/m, it looks like I was already using case-insensitive comparisons everywhere except TemporarySecureMimeContext and WindowsSecureMimeContext, so I've fixed these up to do the same.

**Comment by jstedfast** [comment]
Just released 1.2.4 with this fix.
